File: src/cn/langya/sun/events/impl/MoveEvent.java
package cn.langya.sun.events.impl;

import com.cubk.event.impl.Event;

/**
 * @author LangYa
 * @ClassName MoveEvent
 * @date 2023/12/28 20:26
 * @Version 1.0
 */

public class MoveEvent implements Event {
    public double x;
    public double y;
    public double z;

    public MoveEvent(final double x, final double y, final double z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }

}

File: src/cn/langya/sun/events/MoveEvent.kt
package cn.langya.sun.events


/**
 * @author LangYa
 * @ClassName MoveEvent
 * @date 2023/12/14 20:58
 * @Version 1.0
 */

class MoveEvent(var x: Double,var y: Double,var z: Double): CancellableEvent() {

    fun zero() {
        this.x = 0.0
        y = 0.0
        z = 0.0
    }

    fun zeroXZ() {
        x = 0.0
        z = 0.0
    }

}